scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -521,6 +537,34 @@ public Boolean call() throws Exception {
         }
     }
 
+    public boolean restartInErrorTask(final JobId jobId, final String taskName)
+            throws UnknownJobException, UnknownTaskException {
+        try {
+            if (status.isUnusable()) {
+                return false;
+            }
+
+            return infrastructure.getClientOperationsThreadPool().submit(new Callable<Boolean>() {
+                @Override
+                public Boolean call() throws Exception {
+                    jobs.restartInErrorTask(jobId, taskName);
+                    wakeUpSchedulingThread();
+                    return Boolean.TRUE;
+                }
+            }).get();
+        } catch (ExecutionException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)